app/src/main/java/org/simple/clinic/newentry/PatientEntryEffectHandler.kt

```diff
@@ -78,4 +84,27 @@ object PatientEntryEffectHandler {
       showDateOfBirthIsInFutureError(false)
     }
   }
+
+  private fun savePatientTransformer(
+      patientRepository: PatientRepository,
+      patientRegisteredCount: Preference<Int>,
+      ui: PatientEntryUi,
+      schedulersProvider: SchedulersProvider
+  ): ObservableTransformer<SavePatient, PatientEntryEvent> {
+    return ObservableTransformer { savePatientEffects ->
+      savePatientEffects
+          .map { it.entry }
+          .subscribeOn(schedulersProvider.io())
+          .flatMapCompletable {
+            patientRepository
+                .saveOngoingEntry(it)
+                .observeOn(schedulersProvider.ui())
+                .andThen(Completable.fromAction {
+                  patientRegisteredCount.set(patientRegisteredCount.get().plus(1))
```

Comment:
This is getting set on the UI thread. Should we move this above into a `doOnNext` block instead?

Reply:
I have split this into an event + effect combo in the upcoming PR. Also, doesn't `SharedPreferences` have its own threading model?

Reply:
It does, but only if you use `apply()` instead of `commit()`. I looked at the implementation of the `Preference` library, and it looks like it does. So I guess this is fine for now.